src/library/scala/collection/Iterable.scala

```diff
@@ -481,9 +481,14 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     *  @see [[scala.collection.Iterator]], method `sliding`
     *
     *  @param size the number of elements per group
-    *  @return An iterator producing ${coll}s of size `size`, except the
-    *          last element (which may be the only element) will be truncated
-    *          if there are fewer than `size` elements remaining to be grouped.
+    *  @return An iterator producing ${coll}s of size `size`, except if there
+    *          are between 1 and `size` elements in the original collection.
+    *          In that case, an iterator producing the collection itself as its
+    *          only element is returned.
```

Comment:
I recommend fixing the docs here and on Iterator, where the examples now show the wrong types. Remove the word "truncated", as that implies data was dropped. (iterator.sliding(n).withPartial(false) does imply truncation, by contrast.) I would suggest keeping the return description brief and moving the explanation to the main text, where you can use a few more words. As the examples show, there are two "special cases" where "the size of the collection is less than `size`", viz., nonEmpty resulting in a single element and isEmpty resulting in an empty iterator. And it's not obvious that isEmpty doesn't result in a single element that is empty. A phrase such as "between 1 and size" sounds confusing to me, because it sounds more precise than the simple idea that it expresses.

Reply:
I think I addressed everything here: No more mention of truncation, slightly updated docs of Iterator -- they were already in better shape than those on Iterable IMO. I note the exception for nonempty < size, but leave the empty case explained explicitly only in the doc comment body to keep the returns text shorter. Does this approach work for you?